File: ruby/robot-name/robot_name.rb
class Robot
  attr_reader :name

  def initialize
    @name = generate_name
  end

  def generate_name
    char_name = (0...2).map { (65 + rand(26)).chr }.join
    num_name = (0...3).map { rand(9) }.join
    char_name + num_name
  end

  def reset
    @name = generate_name
  end
end

File: php/robot-name/robot-name.php
<?php

class Robot
{
  var $name;

  public function getName() {
    if(!isset($this->name)) {
      $name = "";

      $a_z = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";

      $name .= $a_z[rand(0, 25)];
      $name .= $a_z[rand(0, 25)];
      $name .= rand(0, 9);
      $name .= rand(0, 9);
      $name .= rand(0, 9);
      $this->name = $name;
    }

    return $this->name;
  }

  public function reset()
  {
    $this->name = null;
  }
}
